dask/array/core.py

```diff
@@ -2660,6 +2660,11 @@ def bincount(x, weights=None, minlength=None):
 
     return Array(dsk, name, chunks, dtype)
 
+@wraps(np.digitize)
+def digitize(a, bins, right=False):
+    bins = np.asarray(bins)
+    dtype = np.digitize([0], bins, right=False).dtype
+    return a.map_blocks(np.digitize, dtype=dtype, bins=bins, right=right)
```

Comment:
We'll need to add this to the imports in `dask/array/__init__.py` and to the API docs in `docs/source/array-api.rst`.